codespan/src/filemap.rs

```diff
@@ -68,6 +92,10 @@ pub struct FileMap {
 }
 
 impl FileMap {
+    pub fn anonymous(src: String) -> FileMap {
+        Self::new(FileName::virtual_(""), src, ByteIndex::from(1))
```

Comment:
When is this used?

Reply:
Just for tests in my case. Maybe we want to make it `fn new(name: FileName, src: String) -> FileMap` though. I think there is merit to be able to construct standalone `FileMap`s in any case.

Reply:
Hum - yeah, I think what I was trying to avoid was invalid spans being created, like codemap does. But maybe I could just give up on that. 🤔

Reply:
It is still easy to create two codemaps and accidentally use spans from one in the other though. Not really a solvable problem unfortunately :(

Reply:
Could we perhaps remove this and just make the `new` constructor public?

Reply:
Done. I am a bit worried about exposing `start: ByteIndex` as it is not obvious that `0` is is not a valid index.  I hid that part of the public API so that it is only possible to create standalone `FileMap`s without `CodeMap`.